src/traces/funnel/attributes.js

```diff
@@ -0,0 +1,85 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var pieAtts = require('../pie/attributes');
+var barAttrs = require('../bar/attributes');
+var lineAttrs = require('../scatter/attributes').line;
+var extendFlat = require('../../lib/extend').extendFlat;
+var Color = require('../../components/color');
+
+module.exports = {
+    x: barAttrs.x,
+    x0: barAttrs.x0,
+    dx: barAttrs.dx,
+    y: barAttrs.y,
+    y0: barAttrs.y0,
+    dy: barAttrs.dy,
+
+    hovertext: barAttrs.hovertext,
+    hovertemplate: barAttrs.hovertemplate,
+
+    textinfo: extendFlat({}, pieAtts.textinfo, {
+        editType: 'plot',
+        flags: ['label', 'text', 'percent initial', 'percent previous', 'percent total', 'value'],
+        description: [
+            'Determines which trace information appear on the graph.'
+        ].join(' ')
+    }),
+
+    text: barAttrs.text,
+    textposition: extendFlat({}, barAttrs.textposition, {dflt: 'auto'}),
+    insidetextanchor: extendFlat({}, barAttrs.insidetextanchor, {dflt: 'middle'}),
+    insidetextrotate: extendFlat({}, barAttrs.insidetextrotate, {dflt: 'none'}),
+    textfont: barAttrs.textfont,
+    insidetextfont: barAttrs.insidetextfont,
+    outsidetextfont: barAttrs.outsidetextfont,
+    constraintext: barAttrs.constraintext,
+
+    cliponaxis: barAttrs.cliponaxis,
+    orientation: barAttrs.orientation,
+
+    offset: barAttrs.offset,
+    width: barAttrs.width,
```

Comment:
`offset` and `width` seems to _work_ correctly, but I'd vote for removing them. I can't think of a scenario where a user would want to set these attributes.

Reply:
TBD cc: @nicolaskruchten @alexcjohnson

Reply:
I can't think of a reasonable use-case for setting these, so let's remove them for now and we can always add them back in later if it turns out there is a killer app for them.

Reply:
Removing sounds good, only thing I can think of is shifting the bars wrt the axis labels, ie you want all the labels at one end of the bars... but I don’t know why you would want that.

Reply:
I think we may keep both. Here is [an example of use](https://codepen.io/MojtabaSamimi/pen/BEeMrz?editors=0010).

Reply:
<b>Hint</b>: click on each section of the `plotly.js` building.

Reply:
`arrayOk` `width` & `offset` are dropped from `funnel` here:
https://github.com/plotly/plotly.js/blob/c29b58d25b1d61749ddeded441420979f8ad8d13/src/traces/funnel/attributes.js#L49-L50 and the tests are adjusted.
As mentioned, `waterfall` may have a similar issue. So if you think that's a bug, please feel free to open an issue. Thanks.

Reply:
Thanks!